src/pathfinder/water_regions.cpp

```diff
@@ -228,6 +210,7 @@ class WaterRegion
 };
 
 std::vector<WaterRegion> _water_regions;
+std::vector<bool> _is_water_region_valid;
```

Comment:
It might be better to use something similar to `Pool::used_bitmap`.

Reply:
I was thinking of abstracting out a generic bitmap class, but honestly `std::vector<bool>` is a good place to start. Make it work before optimizing? :)

Reply:
I'd rather stick to the std::vector<bool>. Mostly because it's the most out-of-the-box solution :)